src/main/controllers/service/ChooseServiceAreaController.ts

```diff
@@ -39,17 +38,59 @@ export class ChooseServiceAreaController {
   }
 
   public async get(req: FactRequest, res: Response) {
-    const data = await this.getServiceData(req, false);
+    const serviceChosen = req.params.service as string;
+    const action = req.params.action as string;
+    const serviceAreasPageData = req.i18n.getDataByLanguage(req.lng).service;
+    const data = await this.getServiceData(serviceChosen, action, serviceAreasPageData,false);
     res.render('service', data);
   }
 
   public async post(req: FactRequest, res: Response) {
+    enum Action {
+      SendDocuments = 'documents',
+      Update = 'update',
+      NotListed = 'not-listed',
+    }
+
+    enum Catchment {
+      National = 'national',
+      Regional = 'regional'
+    }
+
     if (!hasProperty(req.body, 'serviceArea')) {
-      const data = await this.getServiceData(req, true);
-      res.render('service', data);
+      const serviceChosen = req.params.service as string;
+      const action = req.params.action as string;
+      const serviceAreasPageData = req.i18n.getDataByLanguage(req.lng).service;
+      const serviceData = await this.getServiceData(serviceChosen, action, serviceAreasPageData, true);
+      res.render('service', serviceData);
     } else {
-      res.redirect('/services/unknown-service');
+      const action: string = req.params.action as string;
```

Comment:
seems to be used in both if statements, maybe best to declare and initialise it before